www/docs/migration/migrate-from-v9-to-v10.mdx

```diff
@@ -1,257 +1,82 @@
+  resolve({ input }) {
+    return `hello ${input}!`;
+  },
+});
 
-// NEW:
+// v10
 const appRouter = t.router({
   greeting: t.procedure
     .input(z.string())
     .query(({ input }) => `hello ${input}!`),
 });
 ```
 
-### Calling procedures
-
-```ts
-// OLD
-client.query('greeting', 'KATT');
-trpc.useQuery(['greeting', 'KATT']);
-
-// NEW - you'll be able to CMD+click `greeting` below and jump straight to your backend code
-client.greeting.query('KATT');
-trpc.greeting.useQuery('KATT');
-```
-
-### Inferring types
-
-```ts
-// OLD - you had to make multiple complex helper types yourself
-type GreetingInput = InferQueryInput<'greeting'>;
-type GreetingOutput = InferQueryOutput<'greeting'>;
-
-// NEW - same helper for all procedure types which are shipped out of the box
-type GreetingInput = inferProcedureInput<AppRouter['greeting']>;
-type GreetingOutput = inferProcedureOutput<AppRouter['greeting']>;
-```
-
-### Middlewares
-
-Middlewares are now reusable and can be chained, see [middleware](middlewares) docs.
-
-```ts
-// OLD
-const appRouter = trpc
-  .router()
-  .middleware(({ next, ctx }) => {
-    if (!ctx.user) {
-      throw new TRPCError({ code: 'UNAUTHORIZED' });
-    }
-
-    return next({
-      ctx: {
-        ...ctx,
-        user: ctx.user,
-      },
-    });
-  })
-  .query('greeting', {
-    resolve({ input }) {
-      return `hello ${ctx.user.name}!`;
-    },
-  });
-
-// NEW
-const isAuthed = t.middleware(({ next, ctx }) => {
-  if (!ctx.user) {
-    throw new TRPCError({ code: 'UNAUTHORIZED' });
-  }
-
-  return next({
-    ctx: {
-      // <-- auto-spreading old context, modify only what's changed
-      user: ctx.user,
-    },
-  });
-});
+Under the hood, we are unlocking performance improvements, bringing you quality of life enhancements, and creating room for us to build new features in the future.
 
-// Reusable:
-const authedProcedure = t.procedure.use(isAuthed);
+tRPC v10 features a compatibility layer for users coming from v9. `.interop()` allows you to incrementally adopt v10 so that you can continue building the rest of your project while still enjoying v10's new features.
 
-const appRouter = t.router({
-  greeting: authedProcedure.query(({ ctx }) => {
-    return `Hello ${ctx.user.name}!`
-  }),
-});
-```
+:::caution
+There are [a few features that are not supported by `.interop()`](#limitations-of-interop). We expect nearly all of our users to be able to use `.interop()` to migrate their server side code in only a few minutes. If you are discovering that `.interop()` is not working correctly for you, be sure to [check here](#limitations-of-interop).
+:::caution
 
-## Migration path for the 95%
+## Enabling `interop()` on a v9 router
 
-If you are migrating from V9->V10, the transition is pretty simple for the 95% use-cases.
+Turning your v9 router into a v10 router only takes 10 characters. Add `.interop()` to the end of your v9 router...and you're done with your server code!
 
-**1. Add `.interop()`**
```

Comment:
How come you didn't like the 1/2/3/4 guided way of approaching it? In my head, it's easier to follow